test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +105,63 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', {
+        data: readable,
+        dataType: DATA_TYPE_BINARY,
+        metadata: { raw: [], linkRel: { type: new Set() }, profiles: []} as RepresentationMetadata,
```

Comment:
I would drop these 2 lines since they're unrelated to what actually happens and clutter up the test. Just have `{ data: readable } as Representation`. (Same thing below).